### PR TITLE
parameterize AWS backup config

### PR DESCRIPTION
**Pull request by andycmaj**
tested without specifying AWS provider and config... looks good.

```bash
⮀ docker run --name=gitlabbak -it --rm --link dockergitlab_redis_1:redisio --link dockergitlab_postgresql_1:postgresql sameersbn/gitlab:7.8.1 app:rake gitlab:backup:create
Creating SSH2 RSA key; this may take some time ...
Creating SSH2 DSA key; this may take some time ...
Creating SSH2 ECDSA key; this may take some time ...
Creating SSH2 ED25519 key; this may take some time ...
invoke-rc.d: policy-rc.d denied execution of restart.
Running gitlab rake task...
Dumping database ...
Dumping PostgreSQL database gitlabhq_production ... [DONE]
done
Dumping repositories ...
done
Dumping uploads ...
done
Creating backup archive: 1425080961_gitlab_backup.tar ... done
Uploading backup archive to remote storage  ... skipped
Deleting tmp directories ... done
Deleting old backups ... skipping
```

### CODE REVIEW

File: assets/init [hunk 1]
```diff
@@ -43,6 +43,14 @@ GITLAB_BACKUPS=${GITLAB_BACKUPS:-disable}
 GITLAB_BACKUP_TIME=${GITLAB_BACKUP_TIME:-04:00}
 GITLAB_BACKUP_EXPIRY=${GITLAB_BACKUP_EXPIRY:-}
 
+if [ "${GITLAB_BACKUP_PROVIDER}" == "AWS" ]; then
+  echo "Configuring AWS backups for S3 butket: ${GITLAB_BACKUP_AWS_S3_BUCKET:-}"
```
Comment (sameersbn) [inline assets/init]: @andycmaj you have a typo here (butket) and I think you intend to use `${GITLAB_BACKUP_AWS_S3_BUCKET}` and not `${GITLAB_BACKUP_AWS_S3_BUCKET:-}`

Comment (andycmaj) [inline assets/init]: thanks! will fix

**Comment by andycmaj** [comment]
fixed typos

**Comment by sameersbn** [comment]
@andycmaj I just tried out the changes without configuring the AWS backup options and the backup create option fails.

```bash
# docker run --name='gitlab' -it --rm \
--link postgresql:postgresql --link redis:redisio \
--volume=/opt/docker/gitlab:/home/git/data \
--env-file=/opt/docker/gitlab/envvars \
sameersbn/gitlab:latest app:rake gitlab:backup:create
Updating CA certificates...
Running gitlab rake task...
Dumping database ... 
Dumping PostgreSQL database gitlabhq_production ... [DONE]
done
Dumping repositories ...
...
...
done
Dumping uploads ... 
done
Creating backup archive: 1425473216_gitlab_backup.tar ... done
Uploading backup archive to remote storage  ... rake aborted!
ArgumentError:  is not a recognized storage provider
/home/git/gitlab/vendor/bundle/ruby/2.1.0/gems/fog-core-1.21.1/lib/fog/storage.rb:25:in `new'
/home/git/gitlab/lib/backup/manager.rb:42:in `upload'
/home/git/gitlab/lib/backup/manager.rb:29:in `pack'
/home/git/gitlab/lib/tasks/gitlab/backup.rake:16:in `block (3 levels) in <top (required)>'
Tasks: TOP => gitlab:backup:create
(See full trace by running task with --trace)
```

**Comment by andycmaj** [comment]
hmmm ok i'll take a look. what's your env-file look like?

**Comment by sameersbn** [comment]
my env has the following parameters.

```bash
TZ=Asia/Kolkata
DB_NAME=gitlabhq_production
DB_USER=gitlab
DB_PASS=xxxxxxxxxxx
GITLAB_HTTPS=true
GITLAB_HOST=git.example.com
GITLAB_EMAIL=git@example.com
GITLAB_PROJECTS_VISIBILITY=internal
GITLAB_BACKUPS=daily
GITLAB_BACKUP_TIME=13:00
GITLAB_TIMEZONE=Kolkata
SSL_SELF_SIGNED=true
SMTP_ENABLED=true
SMTP_DOMAIN=mail.example.com
SMTP_USER=noreply@example.com
SMTP_PASS=xxxxxx
OAUTH_ALLOW_SSO=true
OAUTH_BLOCK_AUTO_CREATED_USERS=true
OAUTH_GOOGLE_API_KEY=xxxxxx.apps.googleusercontent.com
OAUTH_GOOGLE_APP_SECRET=xxxxxx
OAUTH_GOOGLE_RESTRICT_DOMAIN=example.com
NGINX_MAX_UPLOAD_SIZE=100m
```

**Pull request closed without merging**